elastic/kibana
# [Response Ops][Alerting] Fix null alerts client reference in rule executors

Resolves https://github.com/elastic/kibana/issues/170683

## Summary

This scenario occurs when there is an issue installing alert resources for a context for a rule type that is using the framework alerts-as-data reporting. When the alert resources are not installed, the task runner passes in `null` for the alertsClient into the executors so when the rule executor is using the new alerts client (vs the alert factory), we see a null accessor error.

We only see this in the ES query rule type because all other rule types using the alerts client performs a check at the start of rule execution. This PR adds that check to the ES query rule type, and in all cases, has the executor throw a controlled `AlertsClientError` if this scenario is encountered.

## To Verify
1. Start up ES and Kibana on `main` and create an ES query run
2. Stop Kibana. Switch to this branch. Make a non-additive change to the alerts field map, like 

```
--- a/packages/kbn-alerts-as-data-utils/src/field_maps/alert_field_map.ts
+++ b/packages/kbn-alerts-as-data-utils/src/field_maps/alert_field_map.ts
@@ -51,7 +51,7 @@ export const alertFieldMap = {
     required: false,
   },
   [ALERT_CASE_IDS]: {
-    type: 'keyword',
+    type: 'date',
     array: true,
     required: false,
   },
```

3. Start Kibana. See errors during startup about not being able to install alert resources. See error thrown during rule execution about the alerts client being expected.
